cmds/stty/termios.go

```diff
@@ -0,0 +1,273 @@
+// Copyright 2015-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"reflect"
+	"strconv"
+)
+
+type (
+	tty struct {
+		Ispeed int
+		Ospeed int
+		Row    int
+		Col    int
+		Line   int
+		Min    int
+		Time   int
+
+		Lnext  int
+		Flush  int
+		Intr   int
+		Quit   int
+		Erase  int
+		Kill   int
+		Eof    int
+		Eol    int
+		Eol2   int
+		Swtch  int
+		Start  int
+		Stop   int
+		Susp   int
+		Rprnt  int
+		Werase int
+
+		Brkint  bool
```

Comment:
yeah, it shoulda been a map. The reason is simple: when we Restore the settings, we can do this:
t, err := gtty(0)
read in this json
set the values *that are set in the map* into the t struct
err := stty(0, t)

that way, values we don't care about remain set. We can have short JSON files that set what we care about and leave everything else alone.